File: C++/34. Find First and Last Position of Element in Sorted Array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        vector<int> pos;
        vector<int> fail={-1,-1};
        int n = nums.size();
        int idx=0;
        for(int i = 0; i<n; i++){
            if(nums[i]==target){
                idx=max(i,idx);
                if(pos.empty()){
                    pos.push_back(i);
                }
            }
        }
        if(!pos.empty()){
            pos.push_back(idx);
            return pos;
        }
        return fail;
    }
};

File: Java/34. Find First and Last Position of Element in Sorted Array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] pos = {-1,-1};
        int n = nums.length;
        int idx=0;
        for(int i = 0; i<n; i++){
            if(nums[i]==target){
                if(pos[0]==-1){
                    pos[0]=i;
                }
                idx=Math.max(i,idx);
            }
        }
        if(pos[0]!=-1){
            pos[1]=idx;
        }
        return pos;
    }
}
